src/db/index.js

```diff
@@ -28,4 +29,65 @@ async function runMigrations() {
     }
 }
 
-export { pool, runMigrations };
\ No newline at end of file
+// @ts-check
+
+/**
+ * Manages a PostgreSQL transaction with automatic commit/rollback.
+ */
+export class DbTransaction {
+    /**
+     * @type {Client | null}
+     */
+    #client = null;
+
+    /**
+     * @type {Boolean}
+     */
+    #ended = false;
+
+    /**
+     * Starts a transaction (if there isn't one already) and returns the client.
+     * @returns {Promise<Client>}
+     */
+    async tx() {
+        if (this.#ended) throw new Error("Transaction has ended.");
+
+        if (!this.#client) {
+            this.#client = await pool.connect();
+            await this.#client.query('BEGIN');
+        }
+
+        return this.#client;
+    }
+
+    async rollback() {
+        if (!this.#client) return;
+
+        this.#ended = true;
+
+        try {
+            await this.#client.query('ROLLBACK');
+        } catch (e) {
+            logger.error('Failed to rollback transaction: ', e);
+        }
```

Comment:
I think we should release the client like when committing

Reply:
it happens eventually in the dispose anyways, but it feels clearer to me to do it here
not 100% sure though

Reply:
done :)